common/src/main/java/io/novafoundation/nova/common/utils/formatting/duration/TimeDurationFormatter.kt

```diff
@@ -3,13 +3,33 @@ package io.novafoundation.nova.common.utils.formatting.duration
 import kotlin.time.Duration
 import kotlin.time.Duration.Companion.minutes
 
-class TimeDurationFormatter : BoundedDurationFormatter {
+class TimeDurationFormatter(
+    private val useHours: Boolean = true,
+    private val useMinutes: Boolean = true,
+    private val useSeconds: Boolean = true
+) : BoundedDurationFormatter {
 
     override val threshold: Duration = 1.minutes
 
+    init {
+        if (!useHours && !useMinutes && !useSeconds) throw IllegalArgumentException("At least one of the flags should be true")
+    }
+
     override fun format(duration: Duration): String {
         return duration.toComponents { _, hours, minutes, seconds, _ ->
-            "%02d:%02d:%02d".format(hours, minutes, seconds)
+            val args = listOfNotNull(
+                hours.takeIf { useHours },
+                minutes.takeIf { useMinutes },
+                seconds.takeIf { useSeconds }
+            )
```

Comment:
This can be lazily cached to avoid this comutation on each `format`

Reply:
It's can't be cached since we can pass any duration to this function